feat: Utilise create-discord-bot (#10013)
discordjs/discord.js: apps/guide/src/content/02-getting-started/03-adding-your-bot-to-a-server.mdx

---
title: Adding your bot to a server
category: Getting started
---

# Adding your bot to a server

After you [set up an application](./setting-up-an-application), you'll notice it's not in any servers yet. So, how does that work?

> Before you're able to see your bot in a server, you will need to add it by using a unique invite link.

Reviewer: Bot invites are not unique?

Author: Word-to-word from the current guide—I think this means each invite link is specific to the bot (because of the application id), so it's unique in that sense?

Reviewer: I find that specification (unique) to not add much, if any, value and to be more confusing than necessary

Author: Removed the word.

## Bot invite links

The basic version of one such link looks like this:

<CH.Code lineNumbers={false}>

```
https://discord.com/api/oauth2/authorize?client_id=123456789012345678&permissions=0&scope=bot%20applications.commands
```

</CH.Code>

The structure of the URL is quite simple:

- _`https://discord.com/api/oauth2/authorize`_ is Discord's standard structure for authorizing an OAuth2 application (such as your bot application) for entry to a Discord server.
- _`client_id=...`_ is to specify _which_ application you want to authorize. You'll need to replace this part with your client's id to create a valid invite link.
- _`permissions=...`_ describes the permissions that your bot will request to be granted by default upon joining the server you are adding it to.

> - _`scope=bot%20applications.commands`_ specifies that you want to add this application as a Discord bot with the ability to create slash commands.

Reviewer: ```suggestion
- _`scope=bot`_ specifies that you want to add this application as a Discord bot with a presence in servers and the ability to receive gateway events.
```
those are basically mutually exclusive now, as the `bot` scope implies `applications.commands`. The focus of the `bot` scope is the presence in the server, enabling WS connections

Author: Should we strip `%20applications.commands` from the above URL or explain the inference?

Reviewer: oh, i missed that, yes, can be removed.
if you want to cover the `application.commands` scope, I think an info box could work there
I do not remember if we have folding boxes that can be closed by default, or if I'm spending too much time in obsidian recently. that would be a nice use case, if we have that

Author: Went with removing it for now.

<Alert title="Warning" type="warning">
	If you get an error message saying "Bot requires a code grant", head over to your application's settings and disable
	the "Requires OAuth2 Code Grant" option. You shouldn't enable this option unless you know why you need to.
</Alert>

## Creating and using your invite link

> To create an invite link, head back to your application in the [developer portal](https://discord.com/developers/applications), click on your bot application, and open the OAuth2 page.

Reviewer: ```suggestion
To create an invite link, head back to the [developer portal](https://discord.com/developers/applications), click on your bot application, and open the OAuth2 page.
```

> In the sidebar, you'll find the URL generator. Select the _`bot`_ and _`applications.commands`_ options. Once you select the _`bot`_ option, a list of permissions will appear, allowing you to configure the permissions your bot needs.

Reviewer: ```suggestion
In the sidebar, you'll find the URL generator. Select the _`bot`_ option. Once you select the _`bot`_ option, a list of permissions will appear, allowing you to configure the permissions your bot needs.
```

> Grab the link via the "Copy" button and send it in a channel in Discord. CLick on the link you just sent which should reveal this:

Reviewer: ```suggestion
Grab the link via the "Copy" button and send it to a channel in Discord. Click on the link you just sent which should reveal this:
```

![Bot Authorization page](/assets/bot-auth-page.png)

> Choose the server you want to add it to and click "Authorize". Congratulations! You've successfully added your bot to your Discord server.

Reviewer: ```suggestion
Choose the server you want to add the bot to and click "Authorize". Congratulations! You've successfully added your bot to your Discord server.
```

> At this point, you should have a Discord bot via [create-discord-bot](https://github.com/discordjs/discord.js/tree/main/packages/create-discord-bot) with your .env file populated and your Discord bot in a server. You are now ready to do what you like.

Reviewer: ```suggestion
At this point, you should have a Discord bot you created with [create-discord-bot](https://github.com/discordjs/discord.js/tree/main/packages/create-discord-bot) with your .env file populated and your Discord bot in a server. You are now ready to do what you like.
```
